.busted

```diff
@@ -0,0 +1,37 @@
+-- vim: ft=lua tw=80 sw=2 ts=2
+
+return {
+  _all = {
+    helper = 'test/preload.lua',
+    lpath = 'build/?.lua;runtime/lua/?.lua;./?.lua',
+    output = 'test.busted.outputHandlers.nvim',
+    Xhelper = { 'IS_FUNCTIONAL_TEST' },
+  },
+  ---
+  functional = { ROOT = { 'test/functional' } },
+  ----
+  api = { ROOT = { 'test/functional/api' } },
+  autocmd = { ROOT = { 'test/functional/autocmd' } },
+  core = { ROOT = { 'test/functional/core' } },
+  editor = { ROOT = { 'test/functional/editor' } },
+  exCmds = { ROOT = { 'test/functional/ex_cmds' } },
+  lua = { ROOT = { 'test/functional/lua' } },
+  options = { ROOT = { 'test/functional/options' } },
+  plugin = { ROOT = { 'test/functional/plugin' } },
+  provider = { ROOT = { 'test/functional/provider' } },
+  shada = { ROOT = { 'test/functional/shada' } },
+  terminal = { ROOT = { 'test/functional/terminal' } },
+  treesitter = { ROOT = { 'test/functional/treesitter' } },
+  ui = { ROOT = { 'test/functional/ui' } },
+  vimscript = { ROOT = { 'test/functional/vimscript' } },
+  ---
+  unit = {
+    ROOT = { 'test/unit' },
+    Xhelper = { 'IS_UNIT_TEST' },
+  },
+  ---
+  benchmark = {
+    ROOT = { 'test/benchmark' },
+    Xhelper = { 'IS_BENCHMARK_TEST' },
+  },
+}
```

Comment:
Is this file required for this PR?

Reply:
It's a bit tricky to say one or another, so I'll try to summarize a list of pros and cons

### pros
- debugging: since you need very few, or no arguments at all, to run tests `busted path/to/test`
- ergonomics(1): no need create a [custom argparse](https://github.com//neovim/neovim/blob/b7084fef4c850d0352488b14dcff0f36a7e75e1c/cmake/RunTests.cmake#L35-L40) that determines what is safe to override (`--output`) and what isn't (`--lpath`)
- ergonomics(2): no need to [expose environment variables](https://github.com//neovim/neovim/blob/b7084fef4c850d0352488b14dcff0f36a7e75e1c/cmake/RunTests.cmake#L42-L52) either instead of the aforementioned argparse

### cons
- it may seem that running tests directly with busted works otob, you still need to set some environment variables (this used to be a lot more involved before I've delegated a lot of these responsibilities to [`preload.lua`](https://github.com//kylo252/neovim/blob/cbdd5a9662da6c061755682e58d18416e204c946/test/preload.lua#L136-L166))
- bloat(?): always a consideration when adding yet another file to the repo, although it can actually go under `./test`